### ISSUE TITLE
Windows Nightly blockers

### ISSUE THREAD

**Issue by larsbergstrom** [servo/servo#12125]
The below bugs are the ones that are required to at least have a first linked nightly build of Servo on Windows.
- [x] #11307 - Hot spinning on Windows shutdown
- [x] #11967 - Add Windows MSI packaging and publishing of the nigthtly build
- [x] #11969 - Add an icon to Windows
- [x] #12231 - Upload failing due to bad `s3cmd` invocation
- [x] #12523 - Windows tarball missing files
- [x] #13257 - Very slow to load or just white screen launched from installer
- [x] #13371 - implement font rendering on Windows
- [x] #15255 - Windows servo UI missing elements
- [x] #15428 - Windows nightly hangs when loading pages
- [x] #15698 - Windows Servo renders reddit text squished
- [x] #12937 - Keyboard input dropouts

**Comment by Coder206** [comment]
@larsbergstrom So @waddlesplash believes #11969 should be combined with #11967 (the installer). Can you update the post above?

**Comment by waddlesplash** [comment]
Yeah, it makes the most sense to do those two as one PR.

**Comment by larsbergstrom** [comment]
Sounds good! I'll work with @metajack to merge the two PRs. I'll leave them like this for now, if you don't mind, because we don't have a separate issue for the icon & MSI packaging. I'm hoping we'll close this down *very* early next week.

**Comment by Coder206** [comment]
@larsbergstrom Ok, that's fine by me.

**Comment by perlun** [comment]
@larsbergstrom - any progress thus far? (yeah, I know it's still _very_ early in the week your TZ 😛)

**Comment by larsbergstrom** [comment]
I'm going to go ahead and commit to having a working build up on Wednesday (July 20th) by hook or by crook :-)

I apologize to everybody waiting for this!

**Comment by gaby2300** [comment]
I'll be looking forward to test it and QA it!!!

**Comment by easyinplay** [comment]
i am ready

**Comment by thejustsoul** [comment]
Make if possible Portable mode for Windows, without any installer, and store configs\data in program folder.

**Comment by larsbergstrom** [comment]
It looks like this is going to take a couple more days, but we've made really great progress. Sorry for the delay! There's... a lot to fix :-(

**Comment by Coder206** [comment]
@larsbergstrom I am planning on finishing up (cherry pick) with #11969 on Friday. Then, I would be happy to assist on any other issues blocking Windows.

**Comment by larsbergstrom** [comment]
@Coder206 That sounds great! Once @metajack and I finish looking into & debugging the last remaining issues, the MSI will be up and we'd certainly appreciate having others look at it and help find other bugs before we announce / link it from the downloads page.

**Comment by Coder206** [comment]
@larsbergstrom OK great! Looking forward to getting back into the action!

**Comment by Coder206** [comment]
@larsbergstrom @metajack Happy debugging! :-)

**Comment by larsbergstrom** [comment]
Well, we *finally* got the upload step working and have published something... but it's missing servo.exe, so not particularly useful :-/

I'll work on fixing it ASAP, and the following link should have something more useful tomorrow:
https://download.servo.org/nightly/windows/servo-latest.tar.gz

**Comment by larsbergstrom** [comment]
As an update, @metajack has figured out a good solution to our font loading issues and should have some code to tack on to #12525 to have a tarball that works and does not hot spin at shutdown.

After that lands, we'll work more on getting a working MSI & manifest.

**Comment by gaby2300** [comment]
Thank you all so much for your hard work, I am looking very much forward to
testing the first Windows build!!!


Many thanks!
Best regards,
Gabriela
QA Mozilla Hispano Team Responsible


http://www.mozilla-hispano.org
http://twitter.com/mozilla_hispano
http://facebook.com/mozillahispano

**Comment by BladeMight** [comment]
So, how all going?

**Comment by metajack** [comment]
If the PR will land today, the nightly for tomorrow should include Windows!

**Comment by metajack** [comment]
The PR has landed. The next nightly build should include a Windows package.

**Comment by Coder206** [comment]
@metajack Yay! Great work!

**Comment by larsbergstrom** [comment]
There is now a nightly build of Servo for Windows:
https://download.servo.org/nightly/windows/servo-latest.msi

I don't think we should announce *quite* yet, as:
- Startup is pretty slow
- We still spin one CPU at shutdown
- Keyboard input is dropped

BUT, the basic build & publishing infra and most major blockers are gone! Feel free to play with it and start opening bugs... but please don't announce widely, as the stuff above is a bit embarrassing :-)

**Comment by gaby2300** [comment]
Hello,

Awesome news but I'm afraid double clicking the servo-latest.msi doesn't
work in Windows 7. If I try to install it as an addon, there's a message
saying it won't install because it may be corrupt.
I would greatly appreciate your help as I can't wait to test this!!!!


Many thanks!
Best regards,
Gabriela
<http://facebook.com/mozillahispano>

**Comment by metajack** [comment]
@gaby2300 I just downloaded it (after uninstalling the ones I made yesterday) and it installed and ran fine. Can you try downloading it again? The MD5 sum for servo-latest.msi on my machine is b2fdc1cf82d982cf5dcc2f5e9e161b08. Please compare yours as well.

**Comment by metajack** [comment]
@larsbergstrom Initial startup seems to be slow, but second run it seems normal. I'm not sure what it's doing on the first load that slows it down.

I could not reproduce either of the other two issues you mentioned. Can you file them with steps to reproduce?

**Comment by Coder206** [comment]
@metajack Did you get the Servo icon working in this release?

**Comment by gaby2300** [comment]
Hi Jack, thanks for answering so quickly! I would need to know how to find
the MD5 sum you mention to check it. Thanks!

**Comment by rmsEfx** [comment]
Windows 7 64 bit Rus
> Failed to create the context for the backup "C: \ Program Files \ Mozilla Research \ Servo Tech Demo \ servo.exe.Manifest". Error in manifest or policy file "C: \ Program Files \ Mozilla Research \ Servo Tech Demo \ servo.exe.Manifest" in line 9. The element "compatibility" appears as a child element "urn: schemas-microsoft-com: asm. v1 ^ assembly ", which is not supported by this version of Windows.

**Comment by kdar** [comment]
This is the error I get in Win 7 Pro x64:

```
[Window Title]
C:\Program Files\Mozilla Research\Servo Tech Demo\servo.exe

[Content]
C:\Program Files\Mozilla Research\Servo Tech Demo\servo.exe

The application has failed to start because its side-by-side configuration is incorrect. Please see the application event log or use the command-line sxstrace.exe tool for more detail.


[OK]
```

**Comment by rmsEfx** [comment]
Yes, `Deleting the server.exe.manifest file temporarily fixes this.` it does, and the result - white screen in servo window with this picture ... no reaction from any element in window ...
![srvo](https://cloud.githubusercontent.com/assets/19751507/17189682/cb044916-546c-11e6-9c11-805150f304b1.jpg)

**Comment by gaby2300** [comment]
Please  tell me how can I delete the the server.exe.manifest then!

BTW, if I try the same as Kevin I get exactly the same result

Any idea how to fix this?


​M​
any thanks!
Best regards,
Gabriela
QA Mozilla Hispano Team Responsible

**Comment by gaby2300** [comment]
Jack, the revision number is {647E7ADC-1AE6-4405-9EDC-19C7286E888F}. Would
this be what you meant?

Saludos cordiales,
Gabriela
Responsable QA Mozilla Hispano

Un beso,
Besos,
Saludos!!
Gaby
Responsable QA Mozilla Hispano

Many thanks!
Best regards,
Gabriela
QA Mozilla Hispano Team Responsible


http://www.mozilla-hispano.org
http://twitter.com/mozilla_hispano
http://facebook.com/mozillahispano

**Comment by Coder206** [comment]
@gaby2300 Are you able to find the servo.exe.manifest file?

**Comment by metajack** [comment]
Hmm. The manifest specifically enables support for Windows 7. See https://github.com/servo/servo/blob/master/components/servo/servo.exe.manifest#L11

I'm not sure what else is required :(

**Comment by metajack** [comment]
Ah, I think I figured this out. We are missing the correct namespace for the <compatibility> tag.

**Comment by gaby2300** [comment]
@Coder206, Yes, but I cannot open it. I remember I once took out a file
from it's place to the Desktop, opened it with  Note pad, erased something
and then I returned it to it's place. Do I have to do the same here and
delete server.exe.manifest?

**Comment by gaby2300** [comment]
I have a different id so....

**Comment by rmsEfx** [comment]
To metajack: Re-installed, run again ... same error : could not start the application because it's parallel configuration is wrong ... maybe it's something due to the different specific of 64-bit processor architecture. I do run with Intel processor

**Comment by Coder206** [comment]
@gaby2300 I haven't installed the binary yet, I will be in about 2 hours but I recommend waiting for @metajack's fix as it should correct the issues on Windows 7.

**Comment by Coder206** [comment]
Has anyone run Servo on a Windows 10 machine yet?

**Comment by b-strauss** [comment]
Yes, I have tested it on Windows 10:
- very slow startup
- no program icon
- crashed after 2 minutes on idle

**Comment by gaby2300** [comment]
Not me

**Comment by Coder206** [comment]
@b-strauss Thanks

**Comment by gaby2300** [comment]
I'll wait for that then, thanks!

**Comment by b-strauss** [comment]
@Coder206 oh and window resizing does not actually resize the content.

**Comment by Coder206** [comment]
@gaby2300 When @metajack pushes the changes, I will let you know.

@b-strauss That's odd! I thought I was able to resize Wikipedia content

**Comment by gaby2300** [comment]
@Coder206, I will greatly appreciate it! I am a long time QA tester and I
am very much looking forward to test this!!!

**Comment by b-strauss** [comment]
@Coder206 It happens on every page.

![screenshot](https://cloud.githubusercontent.com/assets/6588177/17192750/2709f8e0-5450-11e6-9a09-183a36ad2047.jpg)

**Comment by Coder206** [comment]
@gaby2300 My pleasure! It looks like a fix (#12627) is already being tested on the Servo computers. I am not sure if this will land in the Windows binary of tomorrow.

@b-strauss Thanks for sharing! That's strange! I never saw that before for more than 3 seconds. (It usually resizes) I'll try it on my system in about 1 hour and a half.

**Comment by gaby2300** [comment]
@Coder206, Then maybe on Friday I could try again and see?

**Comment by Coder206** [comment]
@gaby2300 That should give you ample time after the bug fix should be released from what I see.

**Comment by gaby2300** [comment]
Awesome! I can always try tomorrow as well of course! The worst could
happen would be it doesn't work...

**Comment by metajack** [comment]
@gaby2300 If you want to try now, here's the one I built: https://www.dropbox.com/s/hxlh4aemans42pb/Servo.msi?dl=0

The official ones should be published tomorrow, assuming #12627 lands today.

**Comment by rmsEfx** [comment]
Dropbox version didn't even start ...

**Comment by gaby2300** [comment]
@Jack, I downloaded from Dropbox and installed, thanks! But I just get a
white window open for a few seconds, it closes it's self
​.​

**Comment by Lonelobo** [comment]
on my pc (win10 insiders build x64) on intel i7-5960x 16gb ram first run it consumes 6-8gb of my ram when leading (+- 7sec) then its opens the main window and memory usage goes down to 200mb.

**Comment by gaby2300** [comment]
Hi, should I try again? Thanks!

**Comment by metajack** [comment]
@gaby2300 I haven't managed to investigate your issues. I ordered a copy of VMware so I can test directly on Win7. Hopefully I'll have some more info tomorrow or early next week.

**Comment by gaby2300** [comment]
@Jack, cool thanks! Will you update your results here to be able to try
again?

**Comment by Prunar** [comment]
Im getting 'avformat-57.dll not found' on startup (googling tells me this belongs to ffmpeg?). Im on Win10 Pro x64. Via latest msi link today.

**Comment by staabm** [comment]
Even after quitting the ui window servo consumes loads of memory (10GB+)

**Comment by larsbergstrom** [comment]
Once we get https://github.com/servo/servo/pull/12661, we should have a fix for the missing DLLs in the MSI. I'll push out a new nightly early once that happens :-)

**Comment by Coder206** [comment]
@larsbergstrom Looking forward to seeing the product of your hard work with @metajack :-)

**Comment by KishanV** [comment]
We are waiting for new nightly..

**Comment by larsbergstrom** [comment]
Ugh, now missing libmodplug-1.dll & librtmp-1.dll & libbluray-1.dll & avcodec-57.dll.... :-(

**Comment by gaby2300** [comment]
@Coder206, @Jack, any news or new build to try? Thanks!!!

**Comment by Coder206** [comment]
@gaby2300 Hi! I hope you are doing well. I think you mean @metajack (Mr. Jack Moffitt). Looks like the download page that @larsbergstrom gave us a link to a build that is still throwing errors about missing DLLs, I hope the build will be updated soon.

**Comment by gaby2300** [comment]
@Coder206, thanks for replying so quickly! Do you know if someone will post
here whenever the build is updated?

**Comment by Coder206** [comment]
@gaby2300 Yes, actually, this issue should be closed when all is solved.

**Comment by gaby2300** [comment]
@Coder206 Cool, many thanks!

**Comment by Coder206** [comment]
@gaby2300 😀

**Comment by tapper82** [comment]
Can't wait to test this when it gets a a11y layer.

**Comment by jdm** [comment]
@tapper82 Keep an eye on #4344 in that case.

**Comment by metajack** [comment]
So I've run Servo in win7 and i definitely get browser.html's new tab page and can go to different pages. Unfortunately, I'm running it in VMware, and there are some graphics glitches (the screen turns black after a while, but servo is still working) and it's impossible to know whether that is due to the VM or not. If I turn off graphics acceleration in VMware, then servo refuses to run at all.

I'm not sure how to further debug this :(

@vvuk @glennw is there something we can do on windows that is equivalent to our glxinfo stuff on linux that we've been using to diagnose opengl problems?

**Comment by gaby2300** [comment]
I tried again today in Windows 7. Besides having Windows compatibility
problems (apparently solved by Windows helper but not quite), the program
can't open because the libbluray-1.dll is missing in my machine

​I would greatly appreciate some help!​


Best regards,
Gabriela
QA Mozilla Hispano Team Responsible

**Comment by elahn** [comment]
The program can't start because librtmp-1.dll is missing from your computer. Try reinstalling the program to fix the problem.

Windows 10 with Anniversary Update on real hardware

**Comment by gaby2300** [comment]
I am getting the same as Elahn in Window 7

**Comment by DIGISAGU** [comment]
I am getting the same as Elahn in Windows 10 PRO Anniversary Edition (1607) 64x
on real hardware

**Comment by techtonik** [comment]
After compilation with AppVeyour - https://ci.appveyor.com/project/servo/servo/build/1.0.7057, where to find compiled binaries - `C:/projects/servo/target\\debug` ?

**Comment by cristisilaghi** [comment]
I've tested Servo today and we need more than 4-5 dlls. These are the missing files necessary for Servo to run:
**libbluray-1.dll
libcelt0-2.dll
libffi-6.dll
libgmp-10.dll
libgnutls-30.dll
libgsm.dll
libhogweed-4-2.dll
libidn-11.dll
liblzma-5.dll
libmodplug-1.dll
libmp3lame-0.dll
libnettle-6-2.dll
libogg-0.dll
libopencore-amrnb-0.dll
libopencore-amrwb-0.dll
libopenjp2-7.dll
libopus-0.dll
liborc-0.4-0.dll
libp11-kit-0.dll
librtmp-1.dll
libschroedinger-1.0-0.dll
libspeex-1.dll
libtasn1-6.dll
libtheoradec-1.dll
libtheoraenc-1.dll
libvorbis-0.dll
libvorbisenc-2.dll
libvpx-1.dll
libwavpack-1.dll
libx264-148.dll
libx265.dll
libxml2-2.dll
xvidcore.dll**

For anyone to want to test, here is an archive with those files: http://www.mediafire.com/download/gubbamksgka8oc4/Servo_Missing_Files.zip

**Comment by techtonik** [comment]
> There is now a nightly build of Servo for Windows:
> https://download.servo.org/nightly/windows/servo-latest.msi

![image](https://cloud.githubusercontent.com/assets/515889/17647638/a40dc3ee-6201-11e6-9af2-b4e03a75db7e.png)

What are the commands to build own .msi?

**Comment by SaschaNaz** [comment]
Trying Servo nightly with the DLLs installed only shows blank gray screen. Hmm.

**Comment by DIGISAGU** [comment]
Servo eats all cpu and quater ram. Loading is sow slow

**Comment by DIGISAGU** [comment]
It is so slowwww!!!😭😭😭😭😭😭😭
And it is freezing 😭😭😭😭😭😭😭
plz fix it💖❤💖❤💖❤💖❤

**Comment by JoshTumath** [comment]
I'm very sorry if I come across as rude for saying this, but please only write comments if they are directly helpful in fixing this issue. Remember that a lot of people are receiving notifications for this Issue, and they receive a lot of email spam whenever someone makes a comment about how the build still isn't working. The team have said this is a high priority, and they're doing what they can to get this resolved.

Again, I'm sorry for sounding rude. I just ask that you all bare that in mind. ☺️

**Comment by DIGISAGU** [comment]
It is so slowwww!!!😭😭😭😭😭😭😭
And it is freezing 😭😭😭😭😭😭😭
plz fix it💖❤💖❤💖❤💖❤

**Comment by staabm** [comment]
I guess this wasnt meant to be closed because some of the initial mentioned tickets are still open

**Comment by SaschaNaz** [comment]
Updated set of missing files (2 added, 4 removed):

* libxml2-2.dll
* libgmp-10.dll
* libgnutls-30.dll
* libhogweed-4-1.dll (Added)
* libnettle-6-1.dll (Added)
* liblzma-5.dll
* libgsm.dll
* libmp3lame-0.dll
* libopencore-amrnb-0.dll
* libopencore-amrwb-0.dll
* libopenjp2-7.dll
* libopus-0.dll
* libschroedinger-1.0-0.dll
* libtheoradec-1.dll
* libspeex-1.dll
* libtheoraenc-1.dll
* libvorbis-0.dll
* libvpx-1.dll
* libvorbisenc-2.dll
* libx264-148.dll
* libwavpack-1.dll
* xvidcore.dll
* libx265.dll
* libhogweed-4-2.dll
* libidn-11.dll
* libnettle-6-2.dll
* libp11-kit-0.dll
* libtasn1-6.dll
* liborc-0.4-0.dll
* libogg-0.dll
* libffi-6.dll

**Comment by metajack** [comment]
We can not and will not distribute libmp3lame, libx264, or libx265. There might be others in there that we also can not distribute. libxvidcore looks suspicious for example.

This whole chain of dependencies is being disabled in #12863 until we find a way to bundle a minimal set of ffmpeg dlls or detect and use them at runtime.

**Comment by Eagle3386** [comment]
If legal stuff gets in the way of distributing those DLL files, then provide instructions for how/where to get them at least, please. (Almost) direct links are preferred.

I'd like to test servo, not fight with missing DLLs because of missing detection routines within servo.

**Comment by metajack** [comment]
@Eagle3386 Once #12863 lands and new nightlies are built, none of these will be needed.

We'll turn media metadata back on once we've figured out a way to distribute the correct set.

**Comment by Eagle3386** [comment]
So, I'll have to wait for that then. 😢 Will subscribe to it and looking forward to its landing. 👍

**Comment by SaschaNaz** [comment]
The latest build removed those dependencies! :tada:

**Comment by techtonik** [comment]
Will there be a 32-bit build for older systems like Vista and XP where Chrome doesn't work anymore?

**Comment by elahn** [comment]
> Even after quitting the ui window servo consumes loads of memory (10GB+)
I believe [#11307](https://github.com/servo/servo/issues/11307) - "Hot spinning on Windows shutdown" means when the program is closed, either by clicking the 'X' on the gui window, or shutting down windows (which may be when the issue was first discovered, hence the name). With the latest build on Windows 10 Home x64 with Anniversary Update on real hardware, after closing I get:
![servo 19_08_2016 4_45_13 pm](https://cloud.githubusercontent.com/assets/10919017/17802385/5f9d343e-6633-11e6-8f7c-2dccfafd861f.png)
I made a [video](http://elahn.com.au/uploads/Servo%2019_08_2016%204_41_57%20PM%20(compressed).mp4) of it running. If there are any tests I can run to provide more actionable info, please let me know. I'm happy to install a servo dev environment, if necessary.

**Comment by SaschaNaz** [comment]
@elahn Trying today's nightly does not show that behavior when closing servo after it completely loaded the start screen, but it still spins when closing servo before the complete load.

**Comment by elahn** [comment]
Thanks @SaschaNaz, a new build was released after I downloaded it. I'm in Australia (GMT+10). Looking at the MSI creation date, they're both 19/8/2016 11:29am. Is the "nightly" produced once per day, as the name implies, or after each CI build? If once daily, at what time does it run? (so I can be sure of getting the latest before commenting or filing issues)

**Comment by SaschaNaz** [comment]
@elahn I also have no idea, I just try downloading and compare installer size to see whether it's new.

**Comment by metajack** [comment]
@techtonik I don't think we'll be adding a 32bit build.

**Comment by gaby2300** [comment]
Hi, I just installed the build from https://download.servo.org/nightly/windows/servo-latest.msi
Fortunately no incompatibilities. 
It took several minutes to load a white window with a screenshot containing Servo related information. No way to open new tabs. Is this the expected? Thanks!

**Comment by metajack** [comment]
It sounds like it loaded the servo homepage instead of browser.html. Anyone else having the same issue?

**Comment by gaby2300** [comment]
I think it may be so. The window's title is: http://servo.org/ - Servo

**Comment by elahn** [comment]
No, I just installed the latest build and for me the behaviour is still the same as in this [video](http://elahn.com.au/uploads/Servo%2019_08_2016%206_19_56%20PM%20(compressed).mp4), showing browser.html with most elements not rendered. (Win10)

**Comment by metajack** [comment]
@gaby2300 Inspect the properties of the servo shortcut and see what the command line arguments are. They should be `-w --pref dom.mozbrowser.enabled --pref shell.builtin-key-shortcuts.enabled=false browserhtml\index.html`. If that looks fine, please check that in the same dir as servo.exe there is a browserhtml folder with some HTML and JS and such in it.

**Comment by gaby2300** [comment]
@metajack  The command line for the Servo shortcut seems to be: "C:\Program Files\Mozilla Research\Servo Tech Demo\". I don't think it's what you asked for. 
I don't have a folder named Servo, just a Servo Tech Demo one containing the Servo Tech Demo shortcut only.

**Comment by gaby2300** [comment]
@metajack Please disregard the last part of my earlier post! There is a Sevo Tech Demo folder. It contains the servo.exe file and the browserhtml folder which has a index.html file, and 2 folders components and css
BTW, I run servo.exe again. This time I got a light grey window for a few seconds, then it displayed a white one with a lot of Servo related info and links for a few seconds, then it went white and empty. I am going to leave it open for a while and see if it changes.
Many thanks for your help!

**Comment by techtonik** [comment]
> I don't think we'll be adding a 32bit build.

Is it because Rust is not that cross-platform in the end?

**Comment by Eagle3386** [comment]
I hope it's because Mozilla finally agrees that such old platforms must die.

x64 has so many advantages over its little brother that I can hardly imagine any scenario in which a 32 bit build would be sufficient nowadays, let a alone being better in any way.
Also, XP users (except those rare XP-64 users) would be finally forced to upgrade their Windows system in order to use Servo.

**Comment by ssokolow** [comment]
@techtonik @Eagle3386 

Rust supports 32-bit Win7 as a Tier 1 (MinGW) or Tier 2 (MSVC) platform, so it's probably more "because it's a hassle to target Windows XP and Vista in an automated way and has low return on investment".

As I remember, that's a rough paraphrase of the reasons why Rust's XP targeting support is [Tier 3](https://forge.rust-lang.org/platform-support.html).

**Comment by Eagle3386** [comment]
So, I was partly right - XP sucks and Servo won't support sucking technology. Well done, Mozilla! 👍

**Comment by retep998** [comment]
@ssokolow `i686-pc-windows-msvc` is tier 1. It is only `i586` which is tier 2.

I'm pretty sure Servo already relies significantly on OS functionality which doesn't exist on XP, so I think that ship has sailed unless someone wants to put in the effort to make it all work on XP, which I doubt anyone is willing to do. Besides, as others have pointed out XP is not worth the cost of supporting.

**Comment by ssokolow** [comment]
@retep998 I'm looking at the [platform support page](https://forge.rust-lang.org/platform-support.html) for Rust and I see only three Tier 1 targets for Windows:

```
x86_64-pc-windows-msvc 	✓ 	✓ 	✓ 	64-bit MSVC (Windows 7+)
i686-pc-windows-gnu 	✓ 	✓ 	✓ 	32-bit MinGW (Windows 7+)
x86_64-pc-windows-gnu 	✓ 	✓ 	✓ 	64-bit MinGW (Windows 7+)
```

**Comment by retep998** [comment]
@ssokolow That disagrees with what I see in the [rust book](https://doc.rust-lang.org/book/getting-started.html#tier-1).

**Comment by ssokolow** [comment]
@retep998 Huh. Yours is probably correct then.

**Comment by gaby2300** [comment]
@metajack Please disregard the last part of my earlier post! There is a Sevo Tech Demo folder. It contains the servo.exe file and the browserhtml folder which has a index.html file, and 2 folders components and css
BTW, I run servo.exe again. This time I got a light grey window for a few seconds, then it displayed a white one with a lot of Servo related info and links for a few seconds, then it went white and empty. I am going to leave it open for a while and see if it changes.
Many thanks for your help!

**Comment by codec-abc** [comment]
Hi I tried to run the installer on Windows 10. The installer stop without any message. At this stage, is the installer expected to install servo correctly on Windows 10 (even it the program may have some problems at runtime). By the way is it possible to have a zip version ?

And, is it too soon to update the servo web page to include a link to the windows installer ?

**Comment by metajack** [comment]
As far as I can tell that is how MSI installation works. It installs and then disappears. You should check your start menu and see if it is in there. It should be Mozilla Research->Servo Tech Demo.

@larsbergstrom is there something I'm missing in the installer setup that would cause it to inform people that it finished successfully? Maybe I'm just not used to windows, but I also found this behavior surprising.

**Comment by larsbergstrom** [comment]
@metajack I think we need to add exitdlg.wxs to our `candle.exe` invocation, as here:
https://github.com/torakiki/pdfsam/blob/735703794d1b6274896620f1623cfdfb96ae3651/pdfsam-msi-installer/src/Build.cmd#L13

IIRC individual dialog [steps](http://wixtoolset.org/documentation/manual/v3/wixui/dialog_reference/wixui_dialogs.html) are only added to the steps if they are included in the build steps. I can try this out when I'm back on Monday, too, if that would help :-)

**Comment by codec-abc** [comment]
@metajack @larsbergstrom  Usually the installer let you select the directory where you want to install the software. And generally, they have a checkbox at the end of the installation process to know if you want to run the software right ahead.

**Comment by BladeMight** [comment]
Just [downloaded](https://download.servo.org/nightly/windows/servo-latest.msi) and installed. Running *servo.exe* starts http://servo.org page, so in cmd i do so:
```batch
.\servo.exe .\browserhtml\index.html
```
to run browser.html
First what i noticed:

1. HIGH CPU usage, never been less than ~60%.
2. Memory: Startup(browser.html): [180 MB], any oppened page: [+120-220 MB] => ~300-500 in total.
3. My video driver crashed(i had to shutdown windows forcefully, screen became black(says no VGA)) 2 times after i was using Servo about 15~20 mins.
4. Most visited pages renders correctly.

Windows 10x64, GPU Nvidia GT-220(512 MB), CPU AMD 2x2.5GHz

Hope that would be helpful...

**Comment by gaby2300** [comment]
I tried running servo.exe again today, I got the same kind of windows as before.

Then I tried via cmd and I got a html browser. It didn't' work very well. It was extremely slow, even typing,  sometimes the sites disappeared when I minimize it, others they didn't. 
Site's links didn't work: I tried www.lanacion.com. When clicking on a link nothing happened, nor the pointer turned in to the usual hand.

**Comment by gaby2300** [comment]
I finally managed to browse to one of the above's site link. I couldn't scroll up/down though because there were no scroll bars in the window.

**Comment by gaby2300** [comment]
Yesterday I tried running servo.exe again. I got the same as before.

Then I tried via cmd and I got an html browser. It was extremely slow even for typing. I could just browse  one site at a time, there was no way to open 2 or more tabs. The site's links didn't work, I couldn't scroll them up/down because there were no scroll bars.

**Comment by rwakulszowa** [comment]
Hello.

I've been testing Servo on Windows10 a few times, and here's what I saw:
- there seems to be a huge problem with handling mouse pointer - just moving the mouse consumes lots of CPU. Same goes with clicking the mouse. It happens every time, and - surprisingly - even on blank parts of the New Tab page, where there's literally nothing under the pointer.
_To reproduce: just touch your mouse._
- there is an exception from that - mouse scroll consumes virtually no resources and works fine. That is, if you focus real hard on not moving the pointer :)
- I've also managed to reproduce the hot spinning issue after closing servo, but, after killing the process from the Task Manager, subsequent servo instances don't seem to cause the problem.
_To reproduce: start servo, close the window to cause hot spinning, kill it from the task manager, start servo again and close it_

Obviously, there are also the issues mentioned by others: slow startup time etc.

**Comment by carnoxen** [comment]
hi, I tried to run servo Linux build by windows bash, but something is wrong...
`Failed to create window.: NoBackendAvailable(XOpenDisplayFailed) (thread main, at ../src/libcore/result.rs:783)`

**Comment by SaschaNaz** [comment]
@carnoxen Maybe you should have posted a new issue for that, but you may try this guide: http://windowsreport.com/linux-gui-windows-10-bash/

**Comment by gitmansagu** [comment]
Design is relly good. A lot of cpu usage. Slow startup. Do not eat a lot of ram.
Good work for now. I read that servo is made for 500 fps, but fps is low.
When using servo it automaticly close every 1 min.

**Comment by TYoung86** [comment]
@carnoxen  linux on windows does not have a display component, unless you wire up an x server on windows to host it (there are guides) you probably don't want to play with it. it's not linux on windows _yet_. they've got a ways to go when it comes to exposing linux kernel functionality.

**Comment by aochagavia** [comment]
Just opened #13257 for the blank white screen on Windows 10. Maybe it could be added to the checklist.

**Comment by codec-abc** [comment]
@aochagavia How much time did you wait before taking the screenshot ? I noticed that sometimes the rendering is quite slow to start.

**Comment by aochagavia** [comment]
@codec-abc I just tried again and waited for one or two minutes. Still, everything is white.

**Comment by gaby2300** [comment]
Today I downloaded from the same site as before. Running servo.exe resulted in the same as before.

Then I tried running index.html inside the Servo Tech Demo folder and I got a browser.html tab: 
file:///C:/Program%20Files/Mozilla%20Research/Servo%20Tech%20Demo/browserhtml/index.html. It looked just as a Nightly borwser.

While this time all worked as expected (I could open as many new tabs as I liked, normal speed, scroll bars present, links working), I'd like to know if this is the expected, that is if I am actually running a Servo browser.  Thanks!

**Comment by jdm** [comment]
Opening index.html will open it in your default browser, not Servo.

**Comment by gaby2300** [comment]
Many thanks for answering me so quickly @jdm!

Is the window titled Servo, the parallel browser engine I get when running servo.exe the actual servo one then? It loads a white window containing Servo related links in blue for a few seconds. Then is goes blank. No way to open new tabs. Is this the expected? Thanks!

**Comment by jdm** [comment]
Not expected, no, but seems to be a common problem that other windows testers are experiencing.

**Comment by aochagavia** [comment]
The blank screen is caused by https://github.com/servo/servo/issues/13371. @larsbergstrom do you think we should add this issue to the checklist?

**Comment by xianwenchen** [comment]
Is it possible to provide a zip archive of the Windows build? Thank you!

**Comment by metajack** [comment]
@aochagavia I think it was blank for some people before that change, but yes. I'll add it.

**Comment by larsbergstrom** [comment]
@xianwenchen Do you mean a ZIP archive that packages up the files, or that compresses the MSI? We might be able to make one, but I'd prefer to point people at real installers instead of tarballs, except on platforms where installers require more work than we are currently willing to do (e.g., supporting a variety of linux distros).

**Comment by BladeMight** [comment]
> Do you mean a ZIP archive that packages up the files, or that compresses the MSI?

I think, he meant zip all files, e.g. portable.

**Comment by vvuk** [comment]
I hope we can provide both - zip versions are much easier for developers/testers to work with, without needing to do an install.  They make it possible to have parallel versions available, do bisects, etc.

**Comment by G4udy** [comment]
I've been trying open Servo on Windows7, but close after hundredth seconds. Any idea?

**Comment by Revertron** [comment]
Just tried Servo (from https://download.servo.org/nightly/windows/servo-latest.msi) on Windows 7 - it works almost flawlessly. Just the font rendering is weird - some russian letters become 2 squares :)

**Comment by Ms2ger** [comment]
That may be a known issue with HTML parsing rather than font rendering

**Comment by gitmansagu** [comment]
I start servo on my lepto and it starts very quick. But when I go to diep.io i see yust black screen
![zajeta slika](https://cloud.githubusercontent.com/assets/16504129/19617060/0fd982d4-9826-11e6-8724-ecf9ead3209f.PNG)

**Comment by subscribernamegoeshere** [comment]
todays 2016-10-31 nightly crash right after startup on a windows7 professional x64

msi installer works and installs, and firing up the servo tech demo from e.g. via start menu entry, it loads and draws potentially the main servo window and I even see some kind of search area in the middle and the stilized menu bars on the upper right side I think, rest of the window is completely white, then window freezes and locks down and windows crashreporter appears and apphang error and reports to microsoft.

do we already make use of the windows/microsoft crashreporting and statistic collection for servo binary with a microsoft drwatson/crashreporter account and engine just yet? is anyone looking into these crashes? do windows crashreports and stats for them only work with signed binaries as huge bloat servo.exe 300megs doesnt come with a digital certificate and signature embedded in its binary.

any hints?

**Comment by jdm** [comment]
We do not make use of the automated crash reporting features.

**Comment by Eagle3386** [comment]
@gitmansagu your screenshot looks like XP. Is that true? If so, please bear in mind that a) XP will not be support and b) you should reconsider your OS version and either switch to any Servo-supported OS or at the very least upgrade your OS to Windows 7 as Microsoft's support for Vista ends in April 2017.

**Comment by SaschaNaz** [comment]
@Eagle3386 Probably [an XP theme for Windows 10](http://sagorpirbd.deviantart.com/art/XP-Themes-Final-for-Win10-520345860).

**Comment by Eagle3386** [comment]
@SaschaNaz thanks for the hint and short fun. 👍 I guess, XP is eternal now.. 😆

**Comment by ssokolow** [comment]
To be fair, if I were willing to run Win10 on any of my machines, I'd pick the Zune XP theme (the black one) over that flat design garbage with amateurish widget padding any day.

**Comment by Polyomino** [comment]
Here are some dumps from Windows Error Reporting that might help you guys diagnose crashes.
We haven't indexed pdbs so I just grabbed the ones that came from version 21.21.13.7570 which seems to be the newest.

[2812807c-30ee-46ee-a95d-05b5d8d5335d.zip](https://github.com/servo/servo/files/577033/2812807c-30ee-46ee-a95d-05b5d8d5335d.zip)
[8f5a9048-b1d5-4ffc-a93b-d1f0ecd1e40a.zip](https://github.com/servo/servo/files/577038/8f5a9048-b1d5-4ffc-a93b-d1f0ecd1e40a.zip)
[60afa2d2-79fd-40a0-b1eb-0ee676d9e5c4.zip](https://github.com/servo/servo/files/577039/60afa2d2-79fd-40a0-b1eb-0ee676d9e5c4.zip)
[25ebf8b2-75e3-45ea-b507-54dcd320ea44.zip](https://github.com/servo/servo/files/577040/25ebf8b2-75e3-45ea-b507-54dcd320ea44.zip)

**Comment by gaby2300** [comment]
Yesterday I did the same using Windows 7. The result was a white blank
window opening for a few seconds and that was all. Maybe I did something
wrong though? Is there a newer download to test? Thanks!

**Comment by SaschaNaz** [comment]
It seems #13257 on the OP's list can be marked as resolved and possibly also #11307.

**Comment by brandon-gong** [comment]
just downloaded and ran today's msi file.  Installs cleanly.  Currently the most glaring issues are that there is no icon and also the text renders strange and mismatched.

![image](https://cloud.githubusercontent.com/assets/19562455/20579037/f205ad40-b18f-11e6-8fb7-5c2243c82621.png)

**Comment by firephil** [comment]
https://download.servo.org/nightly/windows/servo-latest.msi

**Comment by lnicola** [comment]
I can't install this, the installer closes (?!) at the "gathering requirements" stage. The window simply disappears. Windows 10 x64.

**Comment by SaschaNaz** [comment]
@lnicola https://github.com/servo/servo/issues/12125#issuecomment-244376266

>As far as I can tell that is how MSI installation works. It installs and then disappears. You should check your start menu and see if it is in there. It should be Mozilla Research->Servo Tech Demo.

**Comment by BladeMight** [comment]
Why don't provide portable zip, instead of msi installer for nightly?

**Comment by ssokolow** [comment]
@SaschaNaz

Normally, you need to opt into unattended mode with a command-line switch. That's where the confusion would be coming in. (Normally, there's at least an "Installation complete" message box when not in unattended mode.)

**Comment by lnicola** [comment]
> As far as I can tell that is how MSI installation works. It installs and then disappears. You should check your start menu and see if it is in there. It should be Mozilla Research->Servo Tech Demo.

That's right, sorry 😞. I looked for it in the Start menu, but Windows takes a couple of seconds to update it and I probably tried too soon. Guess it installs too fast 😄.

Well, it looks like this:

![image](https://cloud.githubusercontent.com/assets/308347/20722890/faf23906-b670-11e6-9519-e0827a367f7b.png)

**Comment by gaby2300** [comment]
It seems something went wrong the last time I installed it. Now I cannot
uninstall it. I would need to delete all the related files in the registry.
Could someone help me please? I would greatly appreciate it!

**Comment by BladeMight** [comment]
>  Now I cannot uninstall it

Install it over again, and then uninstall.

**Comment by gaby2300** [comment]
I cannot install it because the system believes it's already installed

**Comment by BladeMight** [comment]
OK, wait I'll give you log with all installation does to system.

**Comment by BladeMight** [comment]
@gaby2300 here the log:
[Installation_Log.txt](https://github.com/servo/servo/files/620215/Installation_Log.txt)

**Comment by Kein** [comment]
Lol, whose brilliant idea was to pack the compiled binaries into MSI package.

**Comment by gaby2300** [comment]
Fortunately I was able to uninstall it with CC Cleaner, as I wouldn't have
risked deleting so many entries in the registry in case I ruined my Windows
installation!

Unfortunately I am still getting a blank window that only opens for a few
seconds.

**Comment by BladeMight** [comment]
@gaby2300 try running in cmd in servo directory(C:\Program Files\Mozilla Research\Servo Tech Demo):
``` 
servo.exe .\browserhtml\index.html
```
to open browser.html.

**Comment by gaby2300** [comment]
@Maxim, thanks but I get the same, just a blank window for a few seconds

**Comment by BladeMight** [comment]
@gaby2300 well, maybe you should try installing `Visual C++ 2015 x64 redist` if not installed already. Or maybe CCleaner cleanered not enough...
Your `@Maxim` is wrong, it directed not to me, use `@BladeMight` to direct me.

**Comment by gaby2300** [comment]
@BladeMight, many thanks for your help!! Running CCleaner shows Servo Tech Demo again. Besides "Run uninstaller", which I did but seemed not to work, I get 2 other options: "Rename entry" and "Delete entry". Should I try uninstalling again and if it doesn't work then try one of the other 2?

**Comment by BladeMight** [comment]
@gaby2300 deleting entry wont fix anything! Use Run uninstaller and say what is says.

**Comment by BladeMight** [comment]
Try deleting registry folder:
`HKEY_CURRENT_USER\Software\Mozilla Research`
and then run msi again.

**Comment by gaby2300** [comment]
@BladeMight,  I was able to delete it and I installed it again but no changes. Repairing doesn't change anything either.

**Comment by BladeMight** [comment]
@gaby2300 Well, then I don't know why that happens, maybe someone from developers can answer this.

**Comment by gaby2300** [comment]
My Visual C++ 2015 x64 redist version is outdated, do you think updating it may solve the problem? If so, could you please tell me where should I download it from? Thanks!

**Comment by BladeMight** [comment]
@gaby2300 Here:
https://www.microsoft.com/en-us/download/details.aspx?id=53840
pick both and install them.

**Comment by BladeMight** [comment]
Just tested, servo runs OK for me even without any VC++ Redist. @gaby2300 maybe there some other problem.

**Comment by gaby2300** [comment]
@BladeMight I still have the same problem. Regarding the drivers for the video card, my video always works fine so I suppose I have.

**Comment by BladeMight** [comment]
@gaby2300 Well, I have no other opinions why that happennig...

**Comment by BladeMight** [comment]
@gaby2300 According to whole posts I have a clue, that some missing(?) fonts may prevent from normal execute, try restoring your fonts from https://www.youtube.com/watch?v=-YvRa2u58nI

**Comment by gaby2300** [comment]
@BladeMight I'll try that as soon as I can. Many thanks for your help, I greatly appreciate it!

**Comment by gaby2300** [comment]
@BladeMight I'll try that as soon as I can. Many thanks for your help, I greatly appreciate it!

**Comment by gaby2300** [comment]
@BladeMight I watched the video. The fonts are copied from a file in a USB memory, if you now where can I get them, please tell me. Thanks again!

**Comment by BladeMight** [comment]
@gaby2300 There are links under the videos.

**Comment by jonathandturner** [comment]
Should #13257 be checked off the list now?

**Comment by leeoniya** [comment]
...and https://github.com/servo/webrender/issues/617 added

**Comment by jawz101** [comment]
Would it be possible to offer a portable install?  I am restricted from adding icons to my start menu/desktop.

**Comment by BladeMight** [comment]
@jawz101 You could try [extracting the msi](http://superuser.com/questions/307678/how-to-extract-files-from-msi-package).
[Another ways to extract MSI.](https://www.raymond.cc/blog/how-to-view-and-extract-contents-from-a-msi-file/)

**Comment by jawz101** [comment]
Thanks.  That worked.  Well, my office uses SSL Inspection so I guess I'm still out of luck if I can't import its self-signed cert :/  oh well.

**Comment by BladeMight** [comment]
@jawz101 I don't actually understand what you mean :/ SSL? You mean you can't connecnt to SSL sites? Or browser even not starts(oh why would it)? See [my post](https://github.com/servo/servo/issues/12125#issuecomment-244553937) and try to run servo that way.

**Comment by ssokolow** [comment]
@BladeMight I think jawz101 is saying that their office uses an Internet gateway appliance which blocks traffic it can't MitM via a self-signed CA cert.

**Comment by jawz101** [comment]
What he said^.  MITM cert situation.  It bugged me when they turned that on.

**Comment by zticky** [comment]
Tried to install servo but installer crashes on Windows 10.
Unzipped msi but there is no .exe file to start servo

**Comment by BladeMight** [comment]
> Tried to install servo but installer crashes on Windows 10.
Extracted msi but there is no .exe file to start servo

OK, here you go an [extracted working version of Servo](https://mega.nz/#!xI4HxLbA!U96JybAQvCbE-oHAShUqMgPk5mUjGThgiY3Qbv--PXs),  build from 22.12.16. I also added servo-browserhtml.cmd which will start a browser html if running servo.exe did not.

**Comment by lnicola** [comment]
Is it normal for text to look terrible?

**Comment by jdm** [comment]
Yes; native text rendering was added in https://github.com/servo/servo/pull/14153 but it's known to yield poor results at the moment.

**Comment by vvuk** [comment]
That should be largely fixed in webrender upstream - I don't know what
version servo is currently on. I'll check.

**Comment by rbtcollins** [comment]
Can we link the msi on https://download.servo.org/ now?

**Comment by SaschaNaz** [comment]
Shouldn't #14494 be first fixed before nightly release?

**Comment by gaby2300** [comment]
Once downloaded, how do I proceed?

 Running servo-browserhtml.cmd opens a black window and the following
message shows up: "Windows cannot find the "servo" file. Make sure the name
is correct and try again"

I cannot find servo.exe to run. Do I have to install something else before,
aka the normal servo version?

Many thanks for your help!

**Comment by BladeMight** [comment]
> I cannot find servo.exe to run. Do I have to install something else before, aka the normal servo version?

No it is included in archive which you downloaded, you should just extract it in some folder and run `servo.exe` or `servo-browserhtml.cmd`, maybe there was problem with extract?(Or antivirus program)

**Comment by gaby2300** [comment]
Sorry but there is no servo.exe file included in the download.

I cannot extract it. The download I get is a winzip file but right clicking
no context menu shows up so no way to extract.

Running servo-browserhtml.cmd opens a black window and the following
message shows up: "Windows cannot find the "servo" file. Make sure the name
is correct and try again"

**Comment by BladeMight** [comment]
> I cannot extract it.

Use [7-zip](http://7-zip.org/).

**Comment by gaby2300** [comment]
Done!

Regretfully running both servo.exe and servo-browserhtml.cmd just opens a
white window for only a few seconds and closes

**Comment by komawoyo** [comment]
Out of curiosity, do you have another machine you can test on? Download the MSI install and install it on another machine and see if it run. Its running fine for me

**Comment by nikhiljha** [comment]
https://download.servo.org/nightly/windows/servo-latest.msi

I can still reproduce #11307 from this MSI, is this updated?

**Comment by gaby2300** [comment]
Yes, I have a notebook to test, I didn't realize I could try there! I will
do it as soon as possible. Many thanks for your interest!

**Comment by UK992** [comment]
@nikhiljha this link is not updated anymore, use this one: https://download.servo.org/nightly/windows-msvc/servo-latest.msi

**Comment by jonathandturner** [comment]
@UK992 - are we sure that's the latest?  I just tried it out and am seeing that this version doesn't work with most of the links on the startup page:

![servo_hackernews](https://cloud.githubusercontent.com/assets/547158/22663705/8077241a-ed12-11e6-8f37-a859b793df51.png)

![servo_rust_webpage](https://cloud.githubusercontent.com/assets/547158/22663708/842f7c92-ed12-11e6-88ba-7f9a322555f4.png)

![servo_reddit](https://cloud.githubusercontent.com/assets/547158/22663718/92903c40-ed12-11e6-93f8-fb19a4751e03.png)

I also noticed this version will leave zombie processes around when you close the browser (though that may already be a known issue)

**Comment by codec-abc** [comment]
I tried with the same version and it is worst on my side. I have the same problem as @jonathandturner and I have some huge memory leaks. For example, if I open HackerNews then servo start to use all the ram available making my computer unusable.

**Comment by leeoniya** [comment]
seeing same issues here, eating 3GB RAM, crashes randomly.

**Comment by BladeMight** [comment]
6GB RAM, used 1.4GB, running servo gives me message:
>Close programs to prevent information loss:
 servo.exe

Servo window is black, memory used 3GB & cached 1.5GB during servo.exe runtime.

**Comment by batmont** [comment]
required files are missing, the servo.exe cannot be run
vcruntime140.dll
api-ms-win-crt-runtime-l1-1-0.dll
Even after installation http://download.microsoft.com/download/9/3/F/93FCF1E7-E6A4-478B-96E7-D4B285925B00/vc_redist.x64.exe just a blank white window appears and less than a second quickly disappears on windows 7 sp1 x64 (no installed updates)

**Comment by gaby2300** [comment]
I have the same issue!

**Comment by jonathandturner** [comment]
@batmont - can you file an issue so that we can track this down?

**Comment by scineram** [comment]
I am getting this error on 10 Home on first run. Expected?

![servo](https://cloud.githubusercontent.com/assets/16246063/23071050/6857c4a2-f52d-11e6-9d00-cff203cd5831.jpg)

**Comment by andersonn-roberto** [comment]
Got running here without problems.

![image](https://cloud.githubusercontent.com/assets/6409097/23269233/a463b1fe-f9ce-11e6-9b60-319709deb2e1.png)

**Comment by PEmu1** [comment]
Issue #15428 was reported fixed in PR #15681. I can confirm that the latest Windows build is able to load pages without hanging. Stability seems questionable though. I have had it crash multiple times without warning.

**Comment by jonathandturner** [comment]
@scineram did you try to install the missing vc_redist files?

**Comment by BladeMight** [comment]
@scineram About your error code 2755, it must be you have problems with msi installer, could you try to install other msi, and say how it worked? 
Also there is [topic on answers.microsoft](https://answers.microsoft.com/en-us/windows/forum/windows_vista-windows_programs/error-code-2755-windows-installer/c12ffb7a-9984-4daa-8cca-8af6ba30f777) which talks about that error 2755, it may be helpful.

**Comment by koboltzz** [comment]
I had high hopes in servo but month after month I don't see any real progress on windows version. Nearly everything is broken since 6 months.
Why is that?

**Comment by retep998** [comment]
@koboltzz Because not enough developers out there seem to care about Windows. If you want to see progress on Windows support, maybe you could consider submitting some pull requests with improvements.

**Comment by koboltzz** [comment]
I thought servo=mozilla=wantexistence=takefocusonwidleyusedOS=windows ?

**Comment by retep998** [comment]
Sure, but for many of them they only care out of necessity, and since Servo itself is not in Firefox, Mozilla isn't as concerned about how it works on Windows yet. They care more about the bits of Rust that are already in Firefox.

**Comment by leeoniya** [comment]
it *is* rather baffling that mozilla-paid devs dont "care enough" about an OS with 90% market share. telling people they should shut up or contribute doesn't really begin to explain the current disposition towards a half-decent windows experience.

**Comment by jonathandturner** [comment]
Servo is currently a tech demo, not-yet-alpha technology.  We're working to make sure the platform works on desktop platforms, and of course Windows is one of those platforms.  We're also working to ensure that it works on mobile as well.  We, in fact, just hired someone to focus on the Android port.  Ostensibly, I'm also tasked to help with the Windows work, but I'm still getting up to speed.

One of Mozilla's use of the Servo technology is in a more full-featured project: https://wiki.mozilla.org/Quantum  That said, we're still hoping to stand up and polish all the major platforms.

**Comment by KiChjang** [comment]
@koboltzz Can you describe what your problem is when running the current Windows nightly? Lately people have been running into varied success and failures with the Windows build.

@leeoniya AFAIK @retep998 is not a "mozilla-paid dev", and neither am I. Most of the work in Servo is done by volunteers like us, so it isn't too unreasonable to ask for contributions. I think you're exaggerating the tone here when you said "shut up or contribute".

**Comment by leeoniya** [comment]
First, thank you for your work on Servo. As someone who does a lot of open source work and has contributed to Firefox in the past, I'm the last one to complain about anything not being done. However, your statement that "Most of the work in Servo is done by volunteers" doesnt quite pan out, at least not for the past month: https://github.com/servo/servo/pulse/monthly

**Comment by metajack** [comment]
Note that all of the Servo tech in Quantum is almost entirely focused on Windows deployment. The first versions of WebRender to ship in Firefox will be Windows-only I believe. All of those pieces work just fine in Windows and have for a long time.

Please don't assume that because the Servo Tech Demo doesn't work reliably on Windows yet means that the Servo team doesn't care or that Mozilla doesn't care. There has been an enormous amount of effort put into the Windows side to make crash reporting work, debugging work, MSVC native builds work, and all the other pieces that you need for Rust on Windows and Servo tech in Firefox on Windows. Over the last 6 months we've added MSVC builds, stood up Windows CI for Servo and its dependencies, fixed all the bootstrapping, done tons of WebRender and other fixes, and created a new native font backend for Windows. The fact remains the tech demo is in the same sorry state pretty much in terms of user experience, but it's not the case that nothing has been done.

Most of the issues that remain are not fundamental ones, but often require some knowledge of Windows which almost no one in the Servo community has much of. It doesn't help that the debugging experience for Rust on Windows still isn't as good as the experience on macOS or Linux, which means investigating the problems is painful.

We're working on it, and we hope that once we have it working nicely and the development experience less painful, we'll get some community contributors who use Windows primarily to help us keep it moving forward.

**Comment by KiChjang** [comment]
@leeoniya Sure, paid contributors may individually have a higher commit count, but the aggregate of commits by volunteer contributors is at least comparable to the amount of the aggregate from Mozilla devs.

I do not wish to continue this conversation here. Should you have any more questions regarding this, please either find us on #servo at irc.mozilla.org or PM me directly.

**Comment by codec-abc** [comment]
@metajack I was not aware of the MSVC builds. Does that mean that we can compile and debug Servo and windows with only Visual Studio and Python as dependencies now ? About debugging, what are the options to debug on Windows?

**Comment by koboltzz** [comment]
@KiChjang 
Sure:
[https://abload.de/img/2017-03-09_00h42_49o9js2.jpg](url)

**Comment by jonathandturner** [comment]
@codec-abc - I was just playing around with the new VS 2017 Community edition last week.  It seems to debug Servo pretty well out of the box.

I'm currently working on some improvements, so that, eg, you can see std structs like Vec and String a bit better, but it at least does let you step and set breakpoints and watches.

**Comment by BladeMight** [comment]
> I was just playing around with the new VS 2017 Community edition last week. It seems to debug Servo pretty well out of the box.

Does that mean that VS can debug Rust?

**Comment by jonathandturner** [comment]
@BladeMight - yes, it should be able to debug Rust out of the box.  The experience has some rough edges, but you can start up an application, attach to it from the VS debugger, hit pause, and drop into the Rust code.

**Comment by lnicola** [comment]
My experience, with simpler programs than Servo, is that breakpoints more or less work, but VS doesn't know about Rust's `main()` and will often prompt for C++ library source code. I haven't had much success with watches, though.

**Comment by jonathandturner** [comment]
@lnicola - is that with VS 2017 or an earlier version?

**Comment by lnicola** [comment]
@jonathandturner VS 2017

**Comment by jonathandturner** [comment]
@lnicola - interesting.  I've seen a little of that, but I seem to be having more success with it.  I'd be curious to compare notes and see what the difference is.

**Comment by jonathandturner** [comment]
@lnicola - I've started a list of the issues we need to tackle to help polish the debugging experience in VS.  Feel free to add the ones you're hitting to it: https://github.com/rust-lang/rust/issues/40460

**Comment by jackwithoutjill** [comment]
"but often require some knowledge of Windows which almost no one in the Servo community has much of." In other words, this whole project is just a niche for mac and linux techies. Not having a single developer on the team with full experience with an OS that has ~90% market share is just amateurish and a joke.

**Comment by ssokolow** [comment]
No, it just means that the few people who *do* have that knowledge have other, more urgent things to do. (Bugs that affect all platforms equally, deep knowledge that they have to exercise which is even rarer than Windows knowledge, etc.)

Note the "almost" in "almost no one".

**Comment by metajack** [comment]
@jackwithoutjill Your comments are inappropriate. I think you have misunderstood the stage of development that Servo is in. We'll be shipping core components of Servo in Firefox releases later this year, focusing on the Windows platform. However, Servo by itself is not web compatible enough yet that you can use it as a replacement for Chrome or Firefox, and so at this stage we have focused on core functionality that is platform agnostic rather than platform specific functionality. What platform specific functionality we do have now is a direct result of the individuals working on Servo making it usable for themselves during development, not the result of targeting deployment for specific OSes.

**Comment by jonathandturner** [comment]
Removing #15572 from the blockers list.  I haven't been able to repro it except for isolated cases.  Showcase links appear to work correctly.

**Comment by SaschaNaz** [comment]
Please consider adding #14494 to the blockers list.

**Comment by metajack** [comment]
@SaschaNaz Updated to add that.

**Comment by atouchet** [comment]
I don't know if this is concrete enough for its own issue report but I am finding that the main usability problem remaining with the Windows build now is how frequently it crashes. It's nearly impossible to browse for any significant amount of time and even some of the default showcase sites will often crash the browser.

**Comment by jonathandturner** [comment]
@atouchet - across the board there needs to be a fair bit of compatibility and stability improvements.  Some of the stability issues (like the one where you scroll up and down on a webpage and it crashes after a few seconds) are not specific to the Windows build.  I agree it's not particularly great right now.

That said, I'm of the mind that we need to get the Windows nightly out there so more people can try it out and jump in to help.  It's still meant as a tech preview/pre-alpha release.  As long as people understand we're still pre-alpha, I think it's okay to start shipping the nightly.

**Comment by atouchet** [comment]
@jonathandturner I haven't tested the Linux or macOS versions of Servo so I don't have a frame of reference for whether certain issues are exclusive to the Windows build or are more fundamental engine issues that still need to be sorted out. I agree that shipping nightly builds for Windows would be better sooner than later.

**Comment by voracity** [comment]
Not sure if I should note this here: The link on the download page is broken because it says 'lastest' instead of 'latest'.

**Comment by BladeMight** [comment]
@voracity There is already [PR](https://github.com/servo/download.servo.org/pull/21) about this typo.

**Comment by jonathandturner** [comment]
Still have one to go... working on it now

**Comment by mzji** [comment]
Why does #11969 be marked as fixed? It's not merged yet.

**Comment by jonathandturner** [comment]
@mzji - I wonder if it got marked because the Windows build does have a logo but maybe they don't have the permanent solution in place?

**Comment by mzji** [comment]
Currently the MSI installer has an icon, but servo's executable doesn't. I think #11969 could focus on implement this (it requires windows resources).

**Comment by jonathandturner** [comment]
@mzji - maybe I don't understand what you mean?  When I open Servo, I get an icon in the dock:

<img width="45" alt="servo_in_dock" src="https://cloud.githubusercontent.com/assets/547158/24840826/1699f526-1dc9-11e7-8547-e7051954a719.png">

**Comment by atouchet** [comment]
@jonathandturner I have found that the Servo icon in Windows can be kind of funky. Sometimes it will load just fine, sometimes it will load but with noticeable graphical artifacts, and sometimes it just doesn't load at all.

For example here is what I am seeing with the latest nightly build:

![Windows Servo Icon](https://i.imgur.com/Kpbx9ik.png "Windows Servo Icon")

**Comment by jonathandturner** [comment]
@atouchet - if it sometimes loads and sometimes doesn't, it sounds like we have a different issue.  Have you filed a bug for the behaviour you're seeing?  We should probably track it down.

**Comment by atouchet** [comment]
@jonathandturner I haven't filed a bug yet as I wasn't sure if it was just a duplicate of #11969. If you think these are different issues I will document the problem in a bit more detail and file an issue report for it.

Also I'm not sure if you have seen it but I did file an issue for another problem with Servo on the Windows taskbar in #15942. That is kind of annoying but I'm not sure if it is a big enough of an issue to be considered blocking.

**Comment by atouchet** [comment]
@jonathandturner I have filed the taskbar icon issue that I ran into as #16326.

**Comment by mzji** [comment]
@jonathandturner What I mean is the servo main executable (servo.exe) itself doesn't have an icon. By convention windows applications embed an icon in their main executable. Maybe this is the reason of #16326 , but that needs investigate.

**Comment by atouchet** [comment]
#12937 is now done and should be crossed off.

It's not on the list but it would be nice to have #16360 merged to fix the minimization crash issue before Windows nightlies go live. Also I don't know how important the various taskbar icon problems are but they haven't been fixed 100% yet.

**Comment by jonathandturner** [comment]
@atouchet - Was just about to do a build and test these all out.  Since the icon issue seems to happen in release more, I'll build a release build when I test.

#16360 is now merged, so today's build is the first that could possibly become the start of the nightly.  I want to get a clearer picture of how bad the icon thing is, too.

**Comment by gaby2300** [comment]
I tried yet again with a build from https://download.servo.org/nightly/windows-msvc/servo-latest.msi.
I still have the same problem: a blank windows opens for a few seconds and closes.
I would greatly appreciate some help!

**Comment by jonathandturner** [comment]
@gaby2300 - I don't remember if the nightly build gives a backtrace when it fails to run correctly.  Can you try running it from the terminal and seeing what it outputs?

If there isn't anything there, you may need to build a debug build by hand to get the full backtrace.

At the very least, if you file an issue, we can see if we can debug what's going on.

**Comment by gaby2300** [comment]
I tried yet again with a build from https://download.servo.org/nightly/windows-msvc/servo-latest.msi.
I still have the same problem: a blank windows opens for a few seconds and closes.
I would greatly appreciate some help!

**Comment by gaby2300** [comment]
@jonathandturner -  I will run it from the terminal if you tell me how to do so.

I don't know how to build a debug build by hand, I'm no developer, just a very interested and long time QA tester!

File an issue in Github you mean?

**Comment by jonathandturner** [comment]
@gaby2300 - yes, please file a Github issue.  That way we can track it separately rather than trying using this meta issue.  You can tag me in that issue and we can see if we can figure out what's going on.

Thanks!

**Comment by gaby2300** [comment]
@jonathandturner  - Filed issue https://github.com/servo/servo/issues/16393

**Comment by jonathandturner** [comment]
Okay, I gave the current build a test-run.  There are rough edges, to be sure, but that's to be expected of a pre-alpha.  Here's my thought:

I think we're ready for the Windows nightly to go live.

The showcase links work and show something coherent.  Keyboard input works enough for a preview.  We don't have any crashers that happen in the first few seconds of normal use.

There are a couple remaining issues, like the icon not showing up and some users experiencing the odd issue.  But, I'd argue that these don't really impair the thrust of the preview, which is getting Servo into more hands and more potential contributors.  That level of polish we should do, and we will continue to do, but it doesn't feel like a blocking issue for getting more people to open up the Windows nightly and try it.

**Comment by KiChjang** [comment]
Since we have successfully published our Windows nightly build, I'm going to go ahead and close this issue. New issues should be created for any further problems on the Windows nightly.

**Comment by mingwandroid** [comment]
Where can I find the mingw-w64 based build? I prefer my toolchain's source-code to be inspectable.
